internal/pkg/template/diff/diff.go

```diff
@@ -17,12 +19,115 @@ type Node struct {
 	newValue *yaml.Node // Only populated for a leaf node (i.e. that has no child node).
 }
 
-// Parse constructs a diff tree that represent the differences between two YAML documents.
-func Parse(curr, old []byte) (*Node, error) {
-	return &Node{}, nil
-}
-
 // String returns the string representation of the tree stemmed from the diffNode n.
 func (n *Node) String() string {
 	return ""
 }
+
+// From is the YAML document that another YAML document is compared against.
+type From []byte
+
+// Parse constructs a diff tree that represent the differences of a YAML document against the From document.
+func (from From) Parse(to []byte) (*Node, error) {
+	var toNode, fromNode yaml.Node
+	if err := yaml.Unmarshal(to, &toNode); err != nil {
+		return nil, fmt.Errorf("unmarshal current template: %w", err)
+	}
+	if err := yaml.Unmarshal(from, &fromNode); err != nil {
+		return nil, fmt.Errorf("unmarshal old template: %w", err)
+	}
+
+	return (*fromYAML)(&fromNode).parse(&toNode, "")
```

Comment:
I'm okay if you want to keep the _private_ function `parse(from, to, key)` and get rid of the type casting.